native_client/metadata.h

```diff
@@ -0,0 +1,20 @@
+#ifndef METADATA_H_
+#define METADATA_H_
+
+/* Struct for storing an array of character structs.
+   Intended to make it simpler to output metadata
+   through the C API. */
+
+struct Metadata {
+	MetadataItem* items;
+	int num_items;
+};
+
+// Stores each individual character, along with its timing information
+struct MetadataItem {
```

Comment:
Could you precisely describe each component? `timestep` and `start_time`, we should make it obvious about their meaning, their use, their unit.